crates/common/src/add_order.rs

```diff
@@ -439,4 +479,245 @@ _ _: 0 0;
+
+        assert_eq!(add_order_call.config.meta[2], 11);
+
+        assert_eq!(add_order_call.config.meta[3], 12);
+
+        assert_eq!(add_order_call.config.validInputs[0].decimals, 18);
+
+        assert_eq!(add_order_call.config.validInputs[1].decimals, 18);
+
+        assert_eq!(add_order_call.config.validOutputs[0].decimals, 18);
+
+        assert_eq!(add_order_call.post[0].evaluable.bytecode.len(), 4);
+    }
+
+    #[tokio::test]
+    async fn test_add_order_post_action() {
+        let network = Network {
+            name: "test-network".to_string(),
+            rpc: Url::parse(CI_DEPLOY_POLYGON_RPC_URL).unwrap(),
+            chain_id: 137,
+            label: None,
+            network_id: None,
+            currency: None,
+        };
+        let network_arc = Arc::new(network);
+        let deployer = Deployer {
+            network: network_arc.clone(),
+            address: Address::default(),
+            label: None,
+        };
+        let deployer_arc = Arc::new(deployer);
+        let scenario = Scenario {
+            name: "test-scenario".to_string(),
+            bindings: HashMap::new(),
+            runs: None,
+            blocks: None,
+            deployer: deployer_arc.clone(),
+        };
+        let token1 = Token {
+            address: Address::default(),
+            network: network_arc.clone(),
+            decimals: Some(18),
+            label: None,
+            symbol: Some("Token1".to_string()),
+        };
+        let token2 = Token {
+            address: Address::default(),
+            network: network_arc.clone(),
+            decimals: Some(18),
+            label: None,
+            symbol: Some("Token2".to_string()),
+        };
+        let token3 = Token {
+            address: Address::default(),
+            network: network_arc.clone(),
+            decimals: Some(18),
+            label: None,
+            symbol: Some("Token3".to_string()),
+        };
+        let token1_arc = Arc::new(token1);
+        let token2_arc = Arc::new(token2);
+        let token3_arc = Arc::new(token3);
+        let known_vault_id = U256::from(1);
+        let order = Order {
+            inputs: vec![
+                OrderIO {
+                    token: token1_arc.clone(),
+                    vault_id: None,
+                },
+                OrderIO {
+                    token: token2_arc.clone(),
+                    vault_id: Some(known_vault_id),
+                },
+            ],
+            outputs: vec![OrderIO {
+                token: token3_arc.clone(),
+                vault_id: None,
+            }],
+            network: network_arc.clone(),
+            deployer: None,
+            orderbook: None,
+        };
+        let deployment = Deployment {
+            scenario: Arc::new(scenario),
+            order: Arc::new(order),
+        };
+
+        let dotrain = r#"
+some front matter
+---
+#calculate-io
+_ _: 0 0;
+#handle-io
+:;
+#post-add-order
+_ _: 0 0;
```

Comment:
probably good to show something that the add order is intended for, one or both of

- initializing some state
- reverting for governance wrapper reasons and preventing the add

Reply:
these tests are added now